platform/qt/src/qmapboxgl.cpp

```diff
@@ -1086,7 +1103,16 @@ bool QMapboxGL::setPaintProperty(const QString& layer, const QString& property_,
         return false;
     }
 
-    auto result = conversion::setPaintProperty(*layer_, property_.toStdString(), value);
+    conversion::Convertible convertible(value);
+
+    if (value.type() == QVariant::String || value.type() == QVariant::ByteArray) {
```

Comment:
not sure, but it looks like conversion is better to be performed within `conversion::..` code :)

Reply:
It still is - we just need to explicitly pass a `JSValue` object to [Convertible](https://github.com/mapbox/mapbox-gl-native/blob/master/include/mbgl/style/conversion.hpp#L88) in  `conversion::setPaintProperty()` so it know exactly which vtable it should use.